### ISSUE TITLE
Implement UI for new contributions feature

### ISSUE THREAD

**Issue by muffinresearch** [mozilla/addons-frontend#3149]
We're replacing contributions with a simple link. This needs implementing on the new frontend but we will need UX for it. It should be relatively simple in the sense this won't be much more than a link and some context.

**Comment by muffinresearch** [comment]
@eviljeff is the new contrib link and any other relevant data exposed in the API?

**Comment by muffinresearch** [comment]
@pwalm there is a screenshot above, but we could use some guidance on placement - the current mocks have a more advanced (based on old contributions) feature in the side bar.
<img width="264" alt="amo_desktop_-_master_sketch" src="https://user-images.githubusercontent.com/1514/30446756-f94901c6-9981-11e7-82fe-17dec1ac6a7d.png">
So should we do the same placement and adapt the simpler link version to work in the same place?

**Comment by eviljeff** [comment]
... not yet.  There isn't a specific issue in https://github.com/mozilla/addons-server/issues/6069 for it, but it could be done with mozilla/addons-server/issues/6308

**Comment by muffinresearch** [comment]
@jvillalobos not sure if this should be in MVP based on the amount of backlog there is still to go - however, there won't be a contributions feature at all without this. That said, once the data is exposed the implementation is fairly straight-forward.

**Comment by eviljeff** [comment]
Note after the new contributions API is implemented won't return any useful information until developers start filling in the url and I don't believe anyone has been emailed asking them to do that.  (The devhub field and deprecation warning on the old pages have been there for a few weeks but I doubt will have been seen by many users)

**Comment by jvillalobos** [comment]
It's not a primary feature, so it's okay to not make it a blocker. However, it's a sensitive issue since it can affect funding for developers, so we should have it done shortly thereafter.

**Comment by eviljeff** [comment]
The API was implemented in mozilla/addons-server/pull/6436 btw, and the waffle switch was enabled today.

**Comment by muffinresearch** [comment]
@pwalm since this will be a priority for something that we fix post-release would you have some time to provide input on the UX aspects re: placement in the page?

As mentioned above the current mocks are more based on the old contributions feature so we need to adapt the new contributions feature as per the issue description.

**Comment by pwalm** [comment]
Mock and spec:
<img width="868" alt="screen shot 2017-10-26 at 3 34 07 pm" src="https://user-images.githubusercontent.com/6797635/32055886-5f3a7512-ba63-11e7-8960-39494e63b977.png">

Placement:
<img width="341" alt="screen shot 2017-10-26 at 3 35 09 pm" src="https://user-images.githubusercontent.com/6797635/32055896-66445c6a-ba63-11e7-8f4e-c2b975bb32b7.png">

**Comment by pwalm** [comment]
Should we label the button to say "Contribute with [preferred service]"?

**Comment by eviljeff** [comment]
We don't (currently) store the provider names, only the URL, so "contribute
with xxx" will need some API changes. Or constants in frontend and regexs.

**Comment by jvillalobos** [comment]
The name of the service could be pretty long, though. I suggest we keep it as "Contribute" for now.

**Comment by jvillalobos** [comment]
Also note that we're no longer supporting the suggested amount.

**Comment by pwalm** [comment]
Checking out some other sites like GuFundMe or Kickstarter, our language on the button should be a bit more urgent. Can we Make the buttons say "Contribute now"?

**Comment by dessant** [comment]
@pwalm, making the button more urgent and inviting is a great idea. 😋 

The widget is not visible on the placement mockup without scrolling on a 1080p display. Would it be possible to move it somewhere above the fold? Ideally users should learn that contributing is a possibility while they install the add-on, the widget should be in their peripheral field of vision somewhat close to the center of their gaze as they press the install button.

**Comment by jvillalobos** [comment]
"Contribute Now" sounds good to me. As for positioning, let's revisit after we ship this. It's a top request from developers right now.

**Comment by dessant** [comment]
The empty area below the ratings on the right would be perfect for a contribution widget, is something else planned to be placed there?

![screenshot from 2017-11-03 17-54-55](https://user-images.githubusercontent.com/7513068/32383630-2db78a02-c0c1-11e7-9fbe-052786360870.png)

**Comment by willdurand** [comment]
@dessant yes, this space is reserved for more rating information:

<img width="574" alt="screen shot 2017-11-03 at 17 24 09" src="https://user-images.githubusercontent.com/217628/32384632-d289d7e8-c0bb-11e7-80d4-e926e385f7b6.png">

**Comment by dessant** [comment]
@willdurand, yeah, priorities. 😋

**Comment by cadeyrn** [comment]
What does "post release" means exactly for this feature? Since it's the only revenue channel for some add-on developers I see this as high priority and it's really sad that this will not be part of the initial launch.

**Comment by tofumatt** [comment]
Post release means it is an issue to prioritise for after the first site launch. The team is small and has been working very hard to ship what we have so far. Sorry this feature didn’t land in time for the launch but the old PayPal contributions were buggy and we have to improve how we do them. Unfortunately this means we couldn’t get to it in time.

Again, this label means we’re going to try to get to it ASAP. 😊

**Comment by Aris-t2** [comment]
I agree with cadeyrn, "contribute" feature should have the highest priority.
There is no point in launching a new design unless it is **fully** finished and ready to go. Its like buying a car and getting windows or backseats "later".

By the way adding (temporarily) a "Contribute" link inside "More Information" box should be possible even with a "one-man team" considering this bug was opened almost two month ago. We are talking about one single link already present in AMOs database, not figuring out cold fusion. ;-)

**Comment by willdurand** [comment]
@pwalm @jvillalobos is the `$5` "suggested contribution" amount valid? How does that work for users of PayPal in Europe for instance (euro currency)?

**Comment by eviljeff** [comment]
@willdurand 
> Also note that we're no longer supporting the suggested amount.

**Comment by willdurand** [comment]
ah oops. Thanks, problem solved 😅

**Comment by willdurand** [comment]
> Was the positioning changed in the meantime in order to make the option more visible to the user?

Yes!
